### PR TITLE
box2d: don't build unit-tests

### PR DESCRIPTION
**Pull request by neheb**
Fixes CI under Ubuntu 22.04

Signed-off-by: Rosen Penev <rosenp@gmail.com>

### CODE REVIEW

File: subprojects/packagefiles/box2d/meson.build [hunk 1]
```diff
@@ -13,4 +13,7 @@ box2d_dep = declare_dependency(link_with: box2d_lib,
     include_directories: incdir,
     version: meson.project_version())
 
-subdir('unit-test')
+# Upstream has an old versionm of doctest.h which is imcompatible with glibc >=2.34
```
Comment (eli-schwartz) [inline subprojects/packagefiles/box2d/meson.build:16]: Sounds like we should be using *our* doctest subproject instead.

**Review by eli-schwartz — CHANGES_REQUESTED** [review]

File: subprojects/packagefiles/box2d/meson.build (hunk 1 above)
Comment (neheb) [reply subprojects/packagefiles/box2d/meson.build:16]: I....don't know how possible that is. There's a doctest.h in that directory.

Comment (neheb) [reply subprojects/packagefiles/box2d/meson.build:16]: Speficially, the error is:

```
../subprojects/box2d-2.4.1/unit-test/doctest.h:4021:47: error: size of array ‘altStackMem’ is not an integral constant-expression
 4021 |         static char             altStackMem[4 * SIGSTKSZ];
```

Comment (neheb) [reply subprojects/packagefiles/box2d/meson.build]: Huh. It's possible with implicit_include_directories: false . Cool.

**Review by eli-schwartz — APPROVED** [review]

**Pull request merged**